check-oracle.yml

```diff
@@ -0,0 +1,148 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+---
+# check-oracle.yml
+- name: "ORAchk: uninstall, install or run"
+  hosts: all
+  become: true
+  gather_facts: false
+  roles:
+    - role: check-oracle
+
+  tasks:
+
+    - name: Uninstall AHF
+      shell: yes 'Y' | /opt/oracle.ahf/bin/ahfctl uninstall
+      when: (uninstall_ahf | bool)
+      ignore_errors: true
+      failed_when: ahf_uninstall_result.rc != 0 and ahf_uninstall_result.rc is defined
+      register: ahf_uninstall_result
+
+    - block:
+        - name: Create AHF directory
+          file:
+            path: "{{ ahf_extract_path }}"
+            state: directory
+            mode: "0700"
+
+        - name: Create ORAchk directory
+          file:
+            path: "{{ orachk_script_dir }}"
+            state: directory
+            mode: "0700"
+
+        - name: Copy AHF file from Google Storage
+          shell: gsutil cp {{ ORA_SWLIB_AHF_FILENAME }} {{ ahf_extract_path }}
+          args:
+            creates: "{{ ahf_extract_path }}/{{ ORA_SWLIB_AHF_FILENAME | basename }}"
+
+        - name: Unzip AHF file
+          unarchive:
+            src: "{{ ahf_extract_path }}/{{ ORA_SWLIB_AHF_FILENAME | basename }}"
+            dest: "{{ ahf_extract_path }}"
+            remote_src: true
+
+        - name: Verify AHF setup signature
+          shell: openssl dgst -sha256 -verify ./oracle-tfa.pub -signature ./ahf_setup.dat ./ahf_setup
+          args:
+            chdir: "{{ ahf_extract_path }}"
+
+        - name: Ensure perl is installed
+          package:
+            name: perl
+            state: present
+
+        - name: Run AHF setup
+          shell: yes 'Y' | ./ahf_setup -extract -notfasetup
```

Comment:
Does this run AHF in daemon mode?  If so, I'd prefere standalone, if just that it's one less thing to clean up.  The intent here is for a one-off anyway.

Reply:
AHF is not fully installed.  The method used is the one documented that allows running ORAchk in standalone mode.  
`check.oracle --ahf-uninstall` removes the AHF installation.
What is left is the zip file and log file created by orachk.

I'm not sure why this would be considered a one off.  ORAchk may be run repeatedly to determine if any remediation efforts were effective.